src/sdam/topology.rs

```diff
@@ -1,7 +1,7 @@
 use std::{
```

Comment:
This file has gotten kind of long and complicated, so once this PR is merged I think I'll go in and break it down into a few smaller ones. For the purposes of preserving the diff I didn't do that in this PR.